### ISSUE TITLE
Degree day calculation unit handling errors

### ISSUE THREAD

**Issue by tlogan2000** [Ouranosinc/xclim#1789]
### Setup Information

- Xclim version: 0.47.0
- Python version:3.11
- Operating System:limux

### Description

Calculation of growing degree days using different units does not yield the same results : 

I believe that in `cumulative_difference` https://github.com/Ouranosinc/xclim/blob/a1a1a25550218330ea36ed428c058275c11579e2/xclim/indices/generic.py#L862  we should maybe convert the data to the threshold units instead of the opposite
doing this `threshold = convert_units_to(threshold, data)`  means that output units should be set to `'F days'` not '`K days`' that I am getting in my output

### Steps To Reproduce

```
import xarray as xr



import matplotlib.pyplot as plt
from clisops.core import subset
import xclim

ds = xr.tutorial.load_dataset("air_temperature", chunks={})
ds = ds.rename({'air':'tas'}).resample(time='D').mean(dim='time')
ds['tas_F'] = xclim.units.convert_units_to(ds.tas, "degF")
# Single point for the example
ds_pt = ds.sel(lon=-73, lat=44, method='nearest')

out1 = xclim.atmos.growing_degree_days(tas=ds_pt.tas, thresh="5 degC", freq="MS")

out2 = xclim.atmos.growing_degree_days(tas=ds_pt.tas_F, thresh="5 degC", freq="MS")

out3 = xclim.atmos.growing_degree_days(tas=ds_pt.tas_F, thresh="278.15 K", freq="MS")

# Plot figure & tadaa! Everything is the same
fig, ax = plt.subplots(figsize=(12, 5))
fig = out1.plot(label="K and degC", linestyle="-", ax=ax)
fig = out2.plot(label="degF and degC", marker="s", markersize=10, linestyle="none", ax=ax)
fig = out3.plot(label="degF and K", marker="o", linestyle="none", ax=ax)
fig = ax.legend()
ax.set_title("Different input units");

```
![image](https://github.com/Ouranosinc/xclim/assets/22454970/e45d1783-b9da-4bad-86b1-42060f045f84)

### Contribution

- [X] I would be willing/able to open a Pull Request to address this bug.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Zeitsperre** [comment]
This merits a patch release once the fix is merged.

**Comment by tlogan2000** [comment]
@aulemahal  The problem seems to be only with fahrenheit. and I think the fix needs to possibly take place in `to_agg_units`  in the end:

The output gets assigned a cf units attr here 
https://github.com/Ouranosinc/xclim/blob/a1a1a25550218330ea36ed428c058275c11579e2/xclim/core/units.py#L555

but from what I see the cf attrs are assigned directly without actually applying a conversion factor i,e,
`pint2cfunits(1 <Unit('degree_Fahrenheit')> * 1 <Unit('day')>`)  gives `'K d'` as an attr but values are still 'F d'

**Comment by aulemahal** [comment]
Indeed
```python3
import xclim as xc
a = xc.core.units.str2pint('1 °F')
b = xc.core.units.str2pint('1 d')

a * b
# 255.92777777777778 <Unit('day * kelvin')>
```

And this on pint 0.23, so not an issue of the latest version. I think the fix goes in `to_agg_units`.

**Comment by aulemahal** [comment]
AH, I see the issue now. Summoning @coxipi with whom we discussed this thing before.

To avoid treating the "delta temperature" issue we decided to _ignore_ the magnitude difference of the integral. `1 °C * 1 d = 274.15 K d` which has an incorrect magnitude in the context of degree days, but the correct units, so we simply muted the magnitude part.

But `1 °F * 1 d = 256 K d` for pint. It would work if it returned `459.67 R d`, but it doesn't...

Previously we did `(orig_u - orig_u) * freq_u` to circumvent this problem. I guess we'll need something alike when a non-absolute scale is detected.
